### PR TITLE
Feature/migration to dot net v5.0

### PR DESCRIPTION
**Pull request by moattarwork**
The PR contains the required changes to upgrade the framework to .NET 5.0. The framework is now supporting a combination of .NET 5.0 and the libraries are multi-targeted for .NET Standard 2.0 and 2.1.

Closing #76 
Closing #69  
Closing #74 

New Features

- #69 - Adding support for .Net 5
- #76 - Adding support for DI with assembly scanning

Fixes

- #74 Fixed the options to load the correct data from the configuration

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

The upgrade has been tested using the unit and integration tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by davidcassell — COMMENTED** [review]
ConfigurationOptionBuilder.cs conflict. I'll return to the rest of the review later.

File: src/Easify.Configurations/Fluents/ConfigurationOptionBuilder.cs [hunk 1]
```diff
@@ -60,10 +61,15 @@ private ConfigurationOptionBuilder AddOptions<TSection>(string section) where TS
...
-            _services.AddOptions<TSection>(section).ValidateDataAnnotations();
 
+            _services.AddOptions<TSection>().Bind(_configuration.GetSection(section)).ValidateDataAnnotations();
             return this;
         }
+        
+        private static string GetSectionName<TSection>() where TSection : class, new()
+        {
+            var section = typeof(TSection).Name;
+            return section;
```
Comment (LewisPercival) [inline src/Easify.Configurations/Fluents/ConfigurationOptionBuilder.cs:72]: This can be a one liner too
return typeof(TSection).Name;

**Review by LewisPercival — COMMENTED** [review]
Looks good Mo! One comment from me.

I know u said the tests were passing but for me one still fails - should this not be picked up in the CI anyway? are we sure that runs all the tests? CI is actually failing for another reason at the moment
image

File: src/Easify.Configurations/Fluents/ConfigurationOptionBuilder.cs (hunk 1 above)
Comment (moattarwork) [reply src/Easify.Configurations/Fluents/ConfigurationOptionBuilder.cs]: For the failing test, it is not happening on my machine even if I killed the folder and re-cloned fresh so I assume something is wrong in yours

**Review by LewisPercival — APPROVED** [review]
Discussed with Mo, approved.

**Review by davidcassell — APPROVED** [review]

**Pull request merged**